### PR TITLE
.golangci.yml: allow use of sync/atomic

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This PR disables the depguard for `sync/atomic` allowing its use in the datadog-agent.

### Motivation

We introduced this depguard in #12736, in order to prevent use of sync/atomic. The reasons for this can be read about in that ticket, but in short, sync/atomic had issues with alignment and an API which was not intuitive, allowing non-atomic access to atomic values.

Since go1.19, new types have been added to sync/atomic which are auto-aligned and prohibit non-atomic access, similar to the go.uber.org/atomic library we replaced sync/atomic with.

Now that sync/atomic is suitable for use, we should allow its use. Eventually we may wish to migrate away from go.uber.org/atomic entirely, but allowing sync/atomic is a good first step.

### Additional Notes

We may eventually want to update the docs in https://github.com/DataDog/datadog-agent/blob/main/docs/dev/atomics.md to no longer recommend the use of uber's atomics library. For now, allowing the use of `sync/atomic` is sufficient.

### CODE REVIEW

**Review by vboulineau — APPROVED** [review]

**Comment by sgnn7** [comment]
@knusbaum Isn't the alignment problem still there for ARM?

https://pkg.go.dev/sync/atomic#pkg-note-BUG still lists the same caveat as before:

> On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically via the primitive atomic functions (types [Int64](https://pkg.go.dev/sync/atomic#Int64) and [Uint64](https://pkg.go.dev/sync/atomic#Uint64) are automatically aligned). The first word in an allocated struct, array, or slice; in a global variable; or in a local variable (because the subject of all atomic operations will escape to the heap) can be relied upon to be 64-bit aligned.

**Review by chouquette — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
Blind LGTM ! 😎

File: .golangci.yml [hunk 1]
```diff
@@ -85,8 +85,6 @@ linters-settings:
         files:
           - $all
         deny:
-          - pkg: "sync/atomic"
-            desc: "Use go.uber.org/atomic instead; see docs/dev/atomics.md"
```
Comment (ogaca-dd) [inline .golangci.yml:89]: What do you think about removing `go.uber.org/atomic` in order to have a single way to handle atomic values?
Note: go.uber.org/atomic could be even marked as deprecated.

Comment (knusbaum) [reply .golangci.yml:89]: I have no strong opinion. I think it generally makes sense to allow whatever libraries people want unless there's a specific safety or stability concern, which there's not for uber/atomic. I can also see the argument for having everyone use the same atomics package.

**Pull request merged**